de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5518790_sys_gh5125_AddFunction_getc_bpartner_product_vendor.sql

```diff
@@ -0,0 +1,24 @@
+DROP FUNCTION IF EXISTS getc_bpartner_product_vendor(numeric);
+
+create or replace function getc_bpartner_product_vendor(p_m_product_id numeric)
+  returns TABLE(productno character varying, productname character varying, c_bpartner_product_id numeric)
+stable
+language sql
+as $$
+SELECT
+  DISTINCT ON (iscurrentvendor, ProductNo, ProductName)
+  ProductNo,
+  ProductName,
+  C_BPartner_Product_ID
+FROM C_BPartner_Product bpp
+WHERE true
+      AND bpp.M_Product_id = p_M_Product_id
+      AND bpp.usedforvendor = 'Y'
+ORDER BY bpp.iscurrentvendor desc, SeqNo, ProductNo, ProductName
+LIMIT 1
+$$;
+
+comment on function getc_bpartner_product_vendor(p_m_product_id numeric)
```

Comment:
i think if there are two C_BPartner_Product's one with `iscurrentvendor='Y'` and one with `iscurrentvendor='N'`
then it actually returns two records, is that what you want?

Reply:
I have limit 1